tests/test_decorators.py

```diff
@@ -67,3 +71,57 @@ def restricted_temp_func(df: pd.DataFrame,**kwargs) -> pd.DataFrame:
     calculated_allocations = restricted_temp_func(df,allocation_lower_limit=0,allocation_upper_limit=0.25)
     assert (df.loc[0:9, PandasEnum.ALLOCATION.value] == 0).all()
     assert (df.loc[10:19, PandasEnum.ALLOCATION.value] == .10).all()
+
+def test_daily_stop_loss():
+    """Checks the limit allocation restriction"""
+    df = simulated_correlated_equities_4_years_gen()
+
+    # constrain fifty fifty allocation
+    @restrict_allocation
+    def restricted_fifty_fifty(df: pd.DataFrame,**kwargs) -> pd.DataFrame:
+        """Restrict fifty-fifty to limits."""
+        return fifty_fifty(df)
+
+    calculated_allocations = restricted_fifty_fifty(df,loss_limit=0.001)
+    calculated_allocations["pct_chg"]=pct_chg(calculated_allocations.price)
+
+    # Boolean to check if percent change less than -loss limit exist
+    check_if_exist=False
+    for ii_item, pct_change in zip(calculated_allocations["allocation"],calculated_allocations["pct_chg"]):
+        
+        if(pct_change<-0.001):
+            assert ii_item==0.0
+            check_if_exist=True
+        else:
+            assert ii_item==0.5
+    
+    assert check_if_exist
+
+
+def test_multiple_restrictions():
+
+    """Checks the limit allocation restriction"""
+    df = simulated_correlated_equities_4_years_gen()
+
+    # constrain fifty fifty allocation
+    @restrict_allocation
+    def restricted_fifty_fifty(df: pd.DataFrame,**kwargs) -> pd.DataFrame:
+        """Restrict fifty-fifty to limits."""
+        return fifty_fifty(df)
+
+    calculated_allocations = restricted_fifty_fifty(df,allocation_lower_limit=0.0, allocation_upper_limit=0.25,loss_limit=0.001)
+    calculated_allocations["pct_chg"]=pct_chg(calculated_allocations.price)
+
+    # Boolean to check if percent change less than -loss limit exist
+    check_if_exist=False
+    for ii_item, pct_change in zip(calculated_allocations["allocation"],calculated_allocations["pct_chg"]):
+        
+        if(pct_change<-0.001):
+            assert ii_item==0.0
+            check_if_exist=True
+        else:
+            assert ii_item==0.25
```

Comment:
We could check allocations are zero for the days after a loss greater than the loss limit.